src/modules/partition/partition.conf

```diff
@@ -137,7 +137,7 @@ initialSwapChoice: none
 # If nothing is specified, Calamares defaults to "ext4".
 #
 # Names are case-sensitive and defined by KPMCore.
-defaultFileSystemType:  "ext4"
+availableFileSystemTypes:  ["ext4", "btrfs", "f2fs"]
```

Comment:
Does this imply order/preference?

Reply:
That makes more sense to me, yes.

Reply:
If you want to remove the default option then you have to change a lot of logic all over the place,
including the forced usage of ext4 in the case of missing default FS. 
Not to mention by just removing it without deprecation or automagically replace,
all existing downstream calamares configuration packages will break, or do at least very unexpected things.

Just do something like 

git grep defaultFileSystemType
git grep ext4

in the calamares source code.

Reply:
> I don't think so but this should not be like this.
> 
> defaultFileSystemType should remain as is, IOW this is what the installer defaults too,
> and availableFileSystemType should be called something like optionalFileSystemTypes.

Isn't that not the exact request you did [2017](https://github.com/calamares/calamares/issues/847) regarding that topic? Default is the first FS you specify. If none is specified it defaults to Ext4.

Reply:
> If you want to remove the default option then you have to change a lot of logic all over the place,
> including the forced usage of ext4 in the case of missing default FS.
> Not to mention by just removing it without deprecation or automagically replace,
> all existing downstream calamares configuration packages will break, or do at least very unexpected things.
> 
> Just do something like
> 
> git grep defaultFileSystemType
> git grep ext4
> 
> in the calamares source code.

Well deprecation thing is ok. It's not too much places where defaultFileSystemType is used I think I can address them all. I just think that leaving and using two options is a bit dirty.

Reply:
At the time nothing like the partition scheme existed among many other changes in this area.
But what I requested there is an additional list of supported FSes by the distribution.

Reply:
> Well deprecation thing is ok. It's not too much places where defaultFileSystemType is used I think I can address them all. I just think that leaving and using two options is a bit dirty.

Or maybe hide it in the code.. Eg: we remove the configurationMap check for it, then set it to the first fs in the new_option_of_FSes, with a fallback to ext4 if everything fails ( config bad or whatever else ), that way not too 
much logic needs to change.

Reply:
The point is that **existing** configs which set *defaultFileSystem* now don't set anything at all. The configuration (`setConfigurationMap()`) code could check for that key and decide it means a one-element list, but that does need to be spelled out.